Remove broken responsive-tables JS, improving mobile readability & accessibility [ci skip]

milesmcc opened the pull request:
### Summary

Currently, tables in the guides are broken on mobile. The first column (usually the label of the row) is removed by the `responsive-tables.js` script, making them impossible to read. Here's a screenshot of the first table in the getting started guide in Chromium/Chrome/Edge (I've also replicated this issue in iOS Safari):

<img src="https://user-images.githubusercontent.com/3037552/86426691-42dbeb00-bcb6-11ea-9be1-0775d9012513.png" alt="An example of the responsive table script breaking the table" width="300">

Note though that even when the script _does_ work correctly, as it does in Firefox, there's still a readability issue: when the first column of a row is more than one line, the later columns only occupy one line, causing the first rows to be offset from their corresponding values, as illustrated below (see the `Gemfile / Gemfile.lock` cell).

<img src="https://user-images.githubusercontent.com/3037552/86427198-900c8c80-bcb7-11ea-8ec9-424011f88542.png" alt="An example of the table offset being broken" width="300">

Plus, this script is a potentially troublesome from an accessibility standpoint. Because it works by making a complete copy of the table and making one copy absolute (and hiding columns with `.hidden` as necessary), my understanding is that screenreaders would have a hard time properly representing the table to users. 

In short: while this script probably worked quite well in 2012 when it was originally written, browsers have changed since then, and mobile table support has come a long way. **This change simply removes the script, as I believe it's no longer necessary, and it's currently breaking the reading experience on mobile (as shown above).**

Here is what the table from the screenshots above looks like with the script removed (taken on Chromium/Edge):

<img src="https://user-images.githubusercontent.com/3037552/86427412-23de5880-bcb8-11ea-98ba-3985952b8d42.png" alt="Screenshot of the table with the script removed" width="300">

I think this is much more readable than the current version, even without factoring in that the script seems to be broken on Chromium-based browsers and the accessibility concerns.

### Other Information

I'm quite new to Rails, and only discovered this issue because I've been going through the guides on my phone. The guides have been _extremely_ helpful, but unfortunately the reading experience has been difficult with this table issue. Hopefully this fix will make it easier for other Rails-learners!

I've done my best to follow all the contribution guidelines, but apologize in advance if I missed something.

milesmcc committed:
Remove broken responsive-tables JS

Add better responsive CSS for tables in guides

Fix borders & breakpoints for responsive tables

Add table fixes to RTL stylesheets

Use 480px breakpoints, remove outdated CSS

jonathanhefner commented:
Yes, I've found that code to be a nuisance on mobile too!  Larger tables (see the [Routing Guide](https://guides.rubyonrails.org/routing.html) for examples) might benefit from some additional CSS though:

```css
@media only screen and (max-width: 540px) {
  table {
    display: block;
    overflow-x: auto;
  }

  td {
    min-width: 7em;
  }
}
```

What do you think?

milesmcc commented:
That's a great addition, and a good catch. I've added the CSS in the most recent commit of the PR.

The only change I made to what you suggested was to use the breakpoint `639px` instead of `540px` because my understanding is that 640 (well, _less than_ 640) is more standard for mobile devices (I think 540 excludes some devices in landscape mode, though that may have been your intention, in which case I'm happy to change the breakpoint).

jonathanhefner commented:
540px was based on the [smallest table](https://guides.rubyonrails.org/v6.0/active_support_core_extensions.html#to-sentence) I could find, and from eyeballing how the other tables looked in a view port of that width.  My thinking was to avoid forcing `td { min-width: 7em; }` unless necessary.  A 7em column is ~120px wide when including padding and borders, so a 4-column table (like those in the Routing Guide) plus the document padding totals ~525px.  Setting the breakpoint near that lets the browser choose column sizes < 7em when space is less constrained and there isn't risk of making them too small.

And there was another issue that I was side-stepping, but, on second thought, I think we should address.  When a table is narrower than the view port (see [smallest table](https://guides.rubyonrails.org/v6.0/active_support_core_extensions.html#to-sentence) from above), setting `display: block` causes its border box to expand to the full width of the view port, but *not* its `thead` and `tbody`.  We can "fix" this though by removing the [`border` on `table`](https://github.com/rails/rails/blob/fce2d6a78bbe1d3b58edd888f2cdbcb2890e6d11/guides/assets/stylesheets/main.css#L74) and putting it on `thead` and `tbody` instead.  That should preserve the current visual appearance.

jonathanhefner commented on guides/assets/stylesheets/main.css:
```diff
@@ -1,7 +1,7 @@
 /* Main.css */
 /* Created January 30, 2009 */
-/* Modified February 8, 2009
+/* Modified July 3, 2020
```
:rofl:  We've definitely [updated this](https://github.com/rails/rails/commits/master/guides/assets/stylesheets/main.css) since 2009-02-08!  I'm not sure, but perhaps these headers used to be auto-generated?  We can probably just remove them from this and the other CSS files in a separate PR.

milesmcc replied:
Ha -- I guessed as much, but wasn't sure. I'll revert this.

milesmcc commented:
Alright, all set @jonathanhefner -- thanks for the review. Hopefully I interpreted your border suggestion for `thead` and `tbody` correctly!

jonathanhefner commented:
> Hopefully I interpreted your border suggestion for `thead` and `tbody` correctly!

Nailed it!  By the way, since we fixed the table border issue, I'm fine with using a 640px breakpoint if you think that yields better results.  From looking at the rest of the stylesheet, there's also an argument to be made for using 480px, for consistency with the other rules.

Two last things I just noticed:

* We should remove [this CSS](https://github.com/rails/rails/blob/fce2d6a78bbe1d3b58edd888f2cdbcb2890e6d11/guides/assets/stylesheets/main.css#L758-L777) which was added in tandem with `responsive-tables.js` in 65a2977cdd55678d3eab06434625375914511786.

* We should update [`main.rtl.css`](https://github.com/rails/rails/blob/fce2d6a78bbe1d3b58edd888f2cdbcb2890e6d11/guides/assets/stylesheets/main.rtl.css) with the changes we've made here.  I'm not sure how diligent we've been in keeping it in sync with `main.css`, but we can do our best anyway.  :smile: 

After that, if you don't mind [squashing the commits](https://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits), I can give this a merge!

milesmcc commented:
Great, implemented and squashed. I think using `480px` as the breakpoint makes the most sense; better to be consistent with the rest of the stylesheet.

jonathanhefner reviewed (approved):
This looks good to me!  My apologies for not merging this yet, @milesmcc.  I thought I had permissions to merge this because it is a "docs" PR, but it turns out I do not.  I have asked someone to look in on it and give it a merge.  Thank you for your patience! :heart:

milesmcc commented:
No problem @jonathanhefner! Thanks for working with me on getting this PR into shape.
